### PR TITLE
[#490] Add release note for Codemagic CD workflow

### LINKED ISSUE #490: Codemagic: Generate and add "release notes" when distributing new builds on Firebase (staging and production)
## Why

The same idea with #463 , we can improve the Codemagic CD script by adding the Release Note when deploying to Firebase App Distribution. 

## Who Benefits?

Android Developers

### PR DESCRIPTION
**Pull request by lydiasama**
- Closes #490 

## What happened 👀

Add command line script to get the commit message to show as the release note on Codemagic CD workflow

## Insight 📝

- For Staging, we use the latest merge commit message
- For Production, we use the latest 50 merge commit messages from the last tag. 

## Proof Of Work 📹

Staging:

Screenshot 2566-11-09 at 4 56 22 PM

> [!NOTE]
> We need to merge this change to `main` first to see the release note on the Firebase App Distribution.
> But as I test the script, it works correctly in the following image:

Screenshot 2566-11-10 at 11 09 46 AM

### CODE REVIEW

File: codemagic.yaml [hunk 1]
```diff
@@ -81,14 +85,24 @@ workflows:
...
         - push
       branch_patterns:
-        - pattern: main
+        - pattern: 'main'
+    environment:
+      vars:
+        # Increase the clone depth to 50 to cover all commits from the latest tag
+        CM_CLONE_DEPTH: 50
+      groups:
+        - firebase_credentials
```
Comment (lydiasama) [inline codemagic.yaml:94]: We need to add the environment groups here again. It seems to add the `environments > vars` above overrides the environment that we import from [configure_environment](https://github.com/nimblehq/android-templates/pull/558/files#diff-bf0ae6a913219a09775a2f777446c5ebaa55b75a6f39c47bd39ac8ce45b2693aR83)

**Review by ryan-conway — APPROVED** [review]

File: .cicdtemplate/.codemagic/codemagic.yaml [hunk 2]
```diff
@@ -52,13 +53,17 @@ workflows:
...
       branch_patterns:
-        - pattern: develop
+        - pattern: 'develop'
     scripts:
       - *detekt
       - *unit_test
       - name: Build APK for staging
         script: |
           ./gradlew assembleStagingDebug -PversionCode=$BUILD_NUMBER
+      - name: Generating release notes with the latest git commit
```
Comment (chornerman) [inline .cicdtemplate/.codemagic/codemagic.yaml:63]:
```suggestion
      - name: Generate release notes with the latest git commit
```
For consistency, please check the rest as well 🙏🏼

File: .cicdtemplate/.codemagic/codemagic.yaml [hunk 3]
```diff
@@ -78,13 +83,24 @@ workflows:
...
+      groups:
+        - firebase_credentials
+      java: 17
     scripts:
       - *detekt
       - *unit_test
       - name: Build APK for production
         script: |
           ./gradlew assembleProductionDebug -PversionCode=$BUILD_NUMBER
+      - name: Generating release notes with git commits from the latest tag
```
Comment (AVI5HEK) [inline .cicdtemplate/.codemagic/codemagic.yaml:100]: @lydiasama forgot to update in other files. 😆

Comment (lydiasama) [reply .cicdtemplate/.codemagic/codemagic.yaml:100]: Indeed. Thankss @AVI5HEK

**Comment by lydiasama** [comment]
All Updated 🚀  @chornerman @AVI5HEK

**Review by AVI5HEK — APPROVED** [review]

**Pull request merged**